server/middlewares/validate.ts

```diff
@@ -6,7 +6,13 @@ import { APIContext, BaseReq } from "@server/types";
 export default function validate<T extends z.ZodType<BaseReq>>(schema: T) {
   return async function validateMiddleware(ctx: APIContext, next: Next) {
     try {
-      ctx.input = schema.parse(ctx.request);
+      if (!ctx.input) {
+        ctx.input = schema.parse(ctx.request);
+      } else {
+        const { body, query } = schema.parse(ctx.request);
+        ctx.input.body = body;
+        ctx.input.query = query;
+      }
```

Comment:
```suggestion
      ctx.input = {
        ...(ctx.input ?? {}),
        ...schema.parse(ctx.request),
      };
```

How about this? I don't like that we're introducing an additional hardcoding of `body` and `query` here.